### PR TITLE
Fixed Joining Mid-Game Bugging Out Dead Players

### PR DESCRIPTION
**Pull request by 1A3Dev**
- Fixed `isPlayerControlled` being set to `true` on dead players when someone joins mid-game.

This is the cause of the issue where sometimes when you are dead you lose control of the spectator camera and drop from the sky and can walk around on a small platform which is surrounded by void.

### CODE REVIEW

**Comment by 1A3Dev** [comment]
I would like to just say that I hate indentation/spaces...

File: Patches/StartOfRound.cs [hunk 1]
```diff
@@ -5,58 +5,111 @@
...
+            PlayerControllerB pcb = __instance.allPlayerScripts[i];
+            if (pcb.isPlayerControlled && pcb.isPlayerDead) __instance.livingPlayers--;
+        }
+
+        if (NetworkManager.Singleton != null)
+        {
+            for (int j = 0; j < __instance.connectedPlayersAmount + 1; j++)
+            {
+                // Don't set the player as controlled if they are dead.
+                if ((j == 0 || !__instance.allPlayerScripts[j].IsOwnedByServer) && !__instance.allPlayerScripts[j].isPlayerDead)
```
Comment (1A3Dev) [inline Patches/StartOfRound.cs:81]: To make it easier for you to review this and the transpiler are the only additions.

File: Patches/StartOfRound.cs [hunk 2]
```diff
@@ -5,58 +5,111 @@
...
+                {
+                    __instance.allPlayerScripts[j].isPlayerControlled = true;
+                }
+            }
+        }
```
Comment (1A3Dev) [inline Patches/StartOfRound.cs]:
```suggestion
        if (NetworkManager.Singleton != null)
        {
            for (int j = 0; j < __instance.connectedPlayersAmount + 1; j++)
            {
                // Don't set the player as controlled if they are dead.
                if ((j == 0 || !__instance.allPlayerScripts[j].IsOwnedByServer) && !__instance.allPlayerScripts[j].isPlayerDead)
                {
                    __instance.allPlayerScripts[j].isPlayerControlled = true;
                }
            }
        }
```

To make it easier for you to review this and the transpiler are the only additions.

**Comment by anormaltwig** [comment]
Looks good! Thanks for working on this!

**Pull request merged**